reference/service/pulumi-button.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Creating a 'Deploy with Pulumi' Button
+---
+
+The 'Deploy with Pulumi' button provides a way to easily create new Pulumi projects within a web browser. The button can be embedded in README files in GitHub repositories/Gists, blog posts, or other web pages.
+
+For example, the following button can be clicked to configure and create a new empty JavaScript project:
+
+[![Deploy](https://get.pulumi.com/new/button.svg)](https://app.pulumi.com/new?template=https://github.com/pulumi/templates/javascript)
+
+This document describes how you can create Pulumi buttons for your own Pulumi templates, examples, and apps.
+
+## Templates
+
+The Pulumi button works with template projects hosted in public GitHub repositories or Gists. A template is a Pulumi project that has the required `Pulumi.yaml` file, which describes the project. The template project can be in the root of the GitHub repository or within a subdirectory. Multiple projects can be hosted within subdirectories of a single repository.
+
+The `Pulumi.yaml` file can optionally contain a `template` section, which typically includes a `config` section used to specify any required config values for the project. Each config value can have a `description` and `default` value. Config values can also have a `secret` property, which can be set to `true` to indicate the config value is a [secret](../config.html#secrets).
+
+```yaml
+name: my-aws-project
+runtime: nodejs
+description: My AWS project description
+template:
+  config:
+    aws:region:
+      description: The AWS region to deploy into
+      default: us-west-2
+    myAccessToken:
+      description: My access token
+      secret: true
+```
+
+The above snippet includes an `aws:region` config value with a default value of `us-west-2`, as well as a `myAccessToken` config value that is a secret without a default value.
+
+## Testing
+
+You can test your template project with the [Pulumi CLI](https://pulumi.io/quickstart/install.html) or a web browser.
+
+CLI:
+
+```bash
+$ pulumi new https://github.com/pulumi/examples/aws-js-s3-folder
+```
+
+Browser:
+
+```
+https://app.pulumi.com/new?template=https://github.com/pulumi/examples/aws-js-s3-folder
+```
```

Comment:
Might be nice to include a screenshot of the flow - perhaps just a screenshot of this `/new` page for the template you've used as an example here? (with a bit of config available).